### ISSUE TITLE
Register a new URL and Link in Linkcheck whenever a contact is inserted in TinyMCE

### ISSUE THREAD

**Issue by charludo** [digitalfabrik/integreat-cms#3280]
### Motivation

Contacts inserted into TinyMCE in #3169 are picked up by linkcheck *eventually*. But for users, it is not at all apparent why this doesn't happen immediately, potentially leading to confusion when they have inserted a contact in a page, but that page is not shown in the contact's list of pages it is embedded in.

It can also lead to potential inconsistencies:
- insert contact "A" into a page, save
- go to "A"s contact details
- change "A"s name to "B"
- most likely, Linkcheck has not caught the embedding of the contact in the page yet, and so that page will not be updated, still showing "A".

### Proposed Solution

- the creation of a contact through the Contact Form should create a linkcheck `URL` object, where the URL in question is the contact's `full_url`. It should either be checked immediately, or set to be valid during its creation
- inserting the contact into TinyMCE should query that `URL` object, and create a new related `Link` object

Essentially, force linkcheck to recognize the contact.

### Alternatives

None, really

### Related Issues

Blocked by:
- [x] #2959

**Comment by charludo** [comment]
.....is this required? I just implemented a "fix" under the linked branch, but, uhm.....

Could someone else confirm that this is even an issue right now? Because it might just be that my dev environment has `LINKCHECK_DISABLE_LISTENERS=1`, since if I change it to `0`, contacts links are immediately updated 🤦🏼‍♀️ 

Please, if someone else can confirm this would be great

**Comment by osmers** [comment]
@charludo I was just testing the contact functionality and also found the same two problems, i.e. the contact not updating automatically in a page it's inserted in and the page not showing up in the contact's linked pages. So I'd say it's definitely a problem :)

**Comment by charludo** [comment]
> [@charludo](https://github.com/charludo) I was just testing the contact functionality and also found the same two problems, i.e. the contact not updating automatically in a page it's inserted in and the page not showing up in the contact's linked pages. So I'd say it's definitely a problem :)

Laright, thank you for the feedback - where did you experience this? The test cms?

**Comment by osmers** [comment]
@charludo yes, the test cms. I created the POI Herkules in the Testumgebung and created two contacts related to it.

**Comment by charludo** [comment]
> [@charludo](https://github.com/charludo) yes, the test cms. I created the POI Herkules in the Testumgebung and created two contacts related to it.

Oof. Just tried it, and yep, same. Oh no.

**Comment by charludo** [comment]
> > [@charludo](https://github.com/charludo) yes, the test cms. I created the POI Herkules in the Testumgebung and created two contacts related to it.
> 
> Oof. Just tried it, and yep, same. Oh no.
> 
> Edit: I will open a PR with my fix, however: THat fix essentially "disables the disabling" of the linkcheck listeners, and I am rather certain this will lead to problems.

Curiouser and curiouser!

After getting nowhere with my debugging efforts, I tried again on the test cms, and... It worked. (Added a Testkontakt to Baden-Baden, linked it in a page, immediately showed up in the contact form.)

Tried the same for another Landkreis, no issues either.

@osmers could you be so kind and try this out again?  🙈 

I have the following theory: everything works as intended, but if the system is under load, the post-save signals just take a while before they are run. If you run into the same issue again, could you see if waiting a minute or two, then force-refreshing your browser with Strg+F5(just to be save) "fixes" it, i.e. if the page/poi/event where the contact is used shows up in the contact form *then*?

If yes, then I think it needs to be communicated to the Kommunen that it can take a couple of minutes for contact uses to be registered, I'm afraid 😐 

In that case, I think we should *not* merge #3345.

**Comment by charludo** [comment]
> [@charludo](https://github.com/charludo) I just tested this again and it worked without a problem - removing a contact from a page also removed the related entry on the contact page, adding it again, added the page name to the contact overview again and after changing information, the new information was immediately visible on the page where the contact is embedded.

OK, thank you! So, do we ignore this and hope it does not actually become a problem for real users? 😅

**Comment by osmers** [comment]
I think @JoeyStk wants to test this as well - but yeah, I think so and in case there are complaints, we can always add it to the wiki and basically know to tell them that sometimes it just takes some time. If it really happens a lot, we can revisit this issue.

**Comment by charludo** [comment]
> I think [@JoeyStk](https://github.com/JoeyStk) wants to test this as well - but yeah, I think so and in case there are complaints, we can always add it to the wiki and basically know to tell them that sometimes it just takes some time. If it really happens a lot, we can revisit this issue.

Sounds good. I closed the PR.

**Comment by JoeyStk** [comment]
I'm unsure where to write this. Please tell me, if another place is better suited for this: 

In the list of links the shown URL for contact is "/augsburg/contact/5/". If I click on it, I receive a blank page. I think there is `edit` missing is at the end. Then the link should work 🤔

**Comment by JoeyStk** [comment]
> @charludo I was just testing the contact functionality and also found the same two problems, i.e. the contact not updating automatically in a page it's inserted in and the page not showing up in the contact's linked pages. So I'd say it's definitely a problem :)

This was also my experience :/ I also ran into both problems :/

**Comment by osmers** [comment]
> > [@charludo](https://github.com/charludo) I was just testing the contact functionality and also found the same two problems, i.e. the contact not updating automatically in a page it's inserted in and the page not showing up in the contact's linked pages. So I'd say it's definitely a problem :)
> 
> This was also my experience :/ I also ran into both problems :/

@JoeyStk did you test it recently? Then that's a bit annoying :D bcs it means we'll have to inform municipalities that sometimes it takes some time
